src/monodroid/jni/embedded-assemblies.cc

```diff
@@ -736,7 +741,12 @@ EmbeddedAssemblies::typemap_load_file (BinaryTypeMapHeader &header, const char *
 		cur->from = reinterpret_cast<char*>(java_pos);
 
 		uint32_t idx = *(reinterpret_cast<uint32_t*>(java_pos + header.java_name_width));
-		cur->to = reinterpret_cast<char*>(managed_start + (managed_entry_size * idx));
+		if (idx < UINT32_MAX) {
```

Comment:
"Ditto" here, instead of `UINT32_MAX`, we should use a semantically meaningful name, e.g. `INVALID_TARGET_INDEX`.